Client/Assets/LoginsHelper.js

```diff
@@ -599,19 +599,28 @@ function onBlur(event) {
   LoginManagerContent.onUsernameInput(event)
 }
 
-window.addEventListener("load", function(event) {
+var documentBody = document.querySelector('body')
+var observer = new MutationObserver(function(mutations) {
+    findLogins()
+});
+
+observer.observe(documentBody, { attributes: false, childList: true, characterData: false });
+
+function findLogins(event) {
   try {
     for (var i = 0; i < document.forms.length; i++) {
       LoginManagerContent._asyncFindLogins(document.forms[i], { })
```

Comment:
So, I started out with @thebnich's solution, then moved to this one, then couldn't decide which was best and so added the question in the top. On, say, the instagram login page, the mutation observer gets called 6 times, one one of those times does a form get injected. This means that we are doing 6 DOM searches for the form tag (I found that simply searching the added nodes doesn't consistently find the FORM element) over 2 executions of the findLogins. But this might be different in other webpages. 

I'll move to searching for the FORM element though.